### DISCUSSION TITLE
[QUESTION] Access all steel sections available in IDEA StatiCa Connection

### DISCUSSION THREAD

**Discussion by Kar-Myl** [idea-statica/ideastatica-public#507] [Help Q&A] [answered]
Hi,

I am trying to access the names of the available steel sections. Is there some way I could do that using the API or through the grasshopper plugin? 

Thank you in advance!

**Comment by nathanlukeIDEA** [comment]
Hi @Kar-Myl, thanks for the question. It is not currently available, but I am hoping to allow it soon.

**Reply by Kar-Myl** [reply]
Ok. Thank you :)

**Comment by nathanlukeIDEA** [comment, marked as answer]
Hi @Kar-Myl,

I have now added some components (in version 0.1.6) which will allow some searching capabilities of a database. The database is not completely linked so it may have some differences with the app but will help as a guide to available sections.

![image](https://github.com/idea-statica/ideastatica-public/assets/93578548/1acd4cc9-a908-492e-b2fb-3d1750a8bace)



Also when creating a Shape by Name if it is found in the database you will now get a geometry preview of this on the Member. 

![image](https://github.com/idea-statica/ideastatica-public/assets/93578548/8dcae519-a0e4-4663-b665-da8e6aafca1e)

Hopefully this helps.
Cheers
Nathan

**Reply by Kar-Myl** [reply]
Hi @nathanlukeIDEA ,

This looks really good. Thank you! I will test it soon. 😄

**Reply by tbyrk** [reply]
Great addition, @nathanlukeIDEA! I too was struggling with profile names.

One minor issue though. Component logos are swapped.

<img width="263" alt="image" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/ba24b11d-4cb9-4f4c-959c-27817a309564">

**Reply by nathanlukeIDEA** [reply]
Thanks @tbyrk for letting me know. I will fix it. cheers. 

Further to this. I will improve the SAF import to search for database items so items will not have to be parametric to get geometry (if they are found in the database).
